util/dynamicConfig.js

```diff
@@ -1,29 +1,108 @@
+/* eslint-disable no-param-reassign */
 const get = require("get-value");
 const unset = require("unset-value");
 
+function setDynamicConfigValue(event, value, config, field) {
+  value = value.replace("{{", "").replace("}}", "");
+  if (value.includes("||")) {
+    const path = value.split("||")[0].trim();
+    const getFieldVal = get(event, path);
+    if (getFieldVal) {
+      config[field] = getFieldVal;
+      unset(event, path);
+    } else {
+      config[field] = JSON.parse(value.split("||")[1].trim()).toString();
+    }
+  }
+}
+
+function getVal(value, event) {
+  if (Array.isArray(value)) {
+    value.forEach(key => {
+      getVal(key, event);
+    });
+  } else if (typeof value === "object") {
+    Object.keys(value).forEach(obj => {
+      // first checking whether the value is array/object -> in that case we recurse and send that object/array as value
+      if (typeof value[obj] === "object") {
+        getVal(value[obj], event);
+      } // if we encounter string (actual values), then we start configuring it
+      else if (typeof value[obj] === "string") {
+        let val = value[obj];
+        // we need to configure the value here itself else we will not have path
+        if (val && val.startsWith("{{") && val.endsWith("}}")) {
+          val = val.replace("{{", "").replace("}}", "");
+          if (val.includes("||")) {
+            const path = val.split("||")[0].trim();
```

Comment:
```suggestion
           const [pathVal, defaultVal] = val.split("||");
           const path = pathVal.trim();
           ....
           //May be we can add this too(but not sure if we need to add)
           // starts
           if (Array.isArray(defaultVal) || typeof JSON.parse(defaultVal) === 'object') {
               throw new CustomError('Invalid Default Value', 400);
           }
           // ends
           value[obj] = JSON.parse(defaultVal.trim()).toString();
```

Reply:
If not necessary we can skip this because JSON.parse will give error everytime it encouters some string/random value.